### PR TITLE
Samuel Yeo

### PR DESCRIPTION
**Pull request by SK-Sam**
Progress: Completed all functionality + extension of indexes being alphabetical. Also pushed to make the show.html.erb for professors alphabetical

Thoughts: I don't feel great about the tests I created. I felt I wanted to rush a little bit due to the project due soon, also I know I did not quite test for edge cases nor take into account sad paths. I feel I checked off User Stories like a checklist, which I don't want to practice every IC/mid-mod/project. I want to reflect the best of my ability. This was definitely an eye opener as to my ability now and what I'm lacking.

What to work on: 
   - More verbose and wider range tests.
   - joins
   - push more logic into models rather than views
   - more AR, less Ruby
   - style, I should feel more of a need to style a page although it's not in the requirements(don't want to make excuses of "I'm in BE therefore no need")
   
Happy about: 
   - Completing my first IC without much struggle!(based on sheer black and white requirements of User Story)
   - Being able to take on the extension
   - Being able to create tests at least detailed enough to exemplify normal method functionality and perfect scenarios of features
   - Navigating through the application without much inertia as to which file to work on next
   - Creating controllers

### CODE REVIEW

File: app/models/professor.rb [hunk 1]
```diff
@@ -1,3 +1,20 @@
...
+  def full_info
+    "Name: #{name}, Age: #{age}, Specialty: #{specialty}"
+  end
+
+  def order_students_alpha
+    students.order(:name)
+  end
+
+  def students_avg_age
+    #Professor.select('professors.id, students.age').joins(:students).where('professors.id = 1').average(students.age).to_i
```
Comment (timomitchel) [inline app/models/professor.rb:15]: `students.average(:age)`

**Review by timomitchel — COMMENTED** [review]
[Challenge Feedback](https://www.loom.com/share/0f89359080da4c419f84f06ce0cce613)
